tests/test_openapi.py

```diff
@@ -45,3 +53,310 @@ def test_str2bool():
 
     default['ogc_schemas_location'] = '/opt/schemas.opengis.net'
     osl = get_ogc_schemas_location(default)
+
+
+def get_test_file_path(filename):
+    """
+    Helper function to open test file safely
+
+    :param filename: file path
+
+    :returns: corrected file path
+    """
+
+    if os.path.isfile(filename):
+        return filename
+    else:
+        return 'tests/{}'.format(filename)
+
+
+@pytest.fixture()
+def config():
+    """
+    Get the OpenAPI Document configuration
+
+    :returns: OpenAPI configuration YAML dict
+    """
+    with open(get_test_file_path('pygeoapi-test-openapi-config.yml')) as fh:
+        return yaml_load(fh)
+
+
+@pytest.fixture()
+def get_oas_30_(config):
+    """
+    Get the generated OpenAPI 3.0 Document
+
+    :param config: configuration object
+
+    :returns: OpenAPI definition YAML dict
+    """
+    return get_oas_30(config)
+
+
+def test_cql_filters(get_oas_30_, config):
```

Comment:
If you look at other tests you would find many functions to assert minimal unit test cases.
I'd rather splitting this big function to several ones in order to follow the principle of single responsibility and group together assertions with the same scope, like `test_cql_filters_logical_expressions` for logical expressions

Reply:
I will split this big function into multiple smaller functions related to each type of expression assertion.

Reply:
restructured test functions with proper flake8 and linting. please check https://github.com/geopython/pygeoapi/pull/478/commits/c53877cd96c83825b43a19a219b3308daeccf4a0

Reply:
@francbartoli @jorgejesus give a look to this https://github.com/FarheenB/pygeoapi/blob/openapi-cql-extension/tests/test_openapi.py for the above issue